pkg/persistence/redis/redisclient.go

```diff
@@ -0,0 +1,160 @@
+package redis
+
+import (
+"errors"
+"fmt"
+"net/url"
+"strconv"
+"strings"
+"sync"
+"time"
+
+"github.com/FZambia/sentinel"
+"github.com/gomodule/redigo/redis"
+)
+
+var knownPool sync.Map
+var m sync.Mutex
+
+// RedisPoolParam ...
+type RedisPoolParam struct {
```

Comment:
We already have a struct to configure Redis store in `etc.Store`. We could extend it with additional properties. What's more the `etc.Store` respects environment variables to override the default values.

Reply:
I just don't want to touch your old code structure, and can copy this to other project.
Of course if you find this is not quite align with current code, you can refactor this.